wolfcrypt/src/ed448.c

```diff
@@ -528,24 +637,43 @@ int wc_ed448ph_verify_msg(const byte* sig, word32 sigLen, const byte* msg,
 /* Initialize the ed448 private/public key.
  *
  * key  [in]  Ed448 key.
+ * heap [in]  heap pointer to pass to wc_InitShake256().
  * returns BAD_FUNC_ARG when key is NULL
  */
-int wc_ed448_init(ed448_key* key)
+int wc_ed448_init_ex(ed448_key* key, void *heap, int devId)
 {
-    int ret = 0;
+    int ret;
 
-    if (key == NULL) {
-        ret = BAD_FUNC_ARG;
-    }
-    else {
-        XMEMSET(key, 0, sizeof(ed448_key));
+    if (key == NULL)
+        return BAD_FUNC_ARG;
 
-        fe448_init();
-    }
+    XMEMSET(key, 0, sizeof(ed448_key));
+
+#ifdef WOLF_CRYPTO_CB
+    key->devId = devId;
+#else
+    (void)devId;
+#endif
+    key->heap = heap;
+
+    fe448_init();
+
+    ret = wc_InitShake256(&key->sha, heap, INVALID_DEVID);
```

Comment:
We should pass `devId` in here for the third argument.

Reply:
done